apps/andi/lib/andi/input_schemas/datasets.ex

```diff
@@ -114,12 +114,18 @@ defmodule Andi.InputSchemas.Datasets do
       |> StructTools.to_map()
       |> Map.merge(form_changes)
 
-    new_changes = %{technical: technical_changes, business: business_changes, id: dataset_id}
+    user_id =
```

Comment:
I feel like this function could use some refactoring. I think it would be cleaner to make a separate function head that pattern matches out the `user_id` and break this function up